### PR TITLE
test(HACBS-2231): add tests for cleanup-workspace Task

### PR DESCRIPTION
**Pull request by happybhati**
This commit contains tests for cleanup-workspace task
covering all the test scenarios. See [CONTRIBUTING.md](https://github.com/redhat-appstudio/release-service-bundles/blob/main/CONTRIBUTING.md)
for great documentation on how these test work.

Signed-off-by: Happy Bhati <hbhati@redhat.com>

### CODE REVIEW

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        steps:
+          - name: check-result
+            image: quay.io/hacbs-release/release-utils:5b1a1cd9fd68625cab5573ce62e0d87e6f93f341
+            script: |
+              #!/usr/bin/env sh
+
+              INPUT_PATH="$(workspaces.input.path)"
+              REMOVED_DIR="$(workspaces.input.path)/dir-to-be-cleaned-up"
+              # Check if dir-to-be-cleaned-up deleted, it prints non-targeted file contents
+              [ -d "$REMOVED_DIR" ] &&  echo "Something wrong directory $REMOVED_DIR found." ||
```
Comment (johnbieren) [inline catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml:61]: extra space after the `&&`

**Review by johnbieren — COMMENTED** [review]
Your commit title is wrong. It should be like feat(HACBS-2231): ....

**Comment by johnbieren** [comment]
The commit title is still wrong and it needs rebase

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      taskSpec:
+        workspaces:
+          - name: input
+        steps:
+          - name: check-result
+            image: quay.io/hacbs-release/release-utils:5b1a1cd9fd68625cab5573ce62e0d87e6f93f341
+            script: |
+              #!/usr/bin/env sh
+
+              INPUT_PATH="$(workspaces.input.path)"
```
Comment (mmalina) [inline catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml:58]: This seems a bit odd that one subdir has a dedicated variable and the other not. I would suggest just creating the two (and no `INPUT_PATH`):
```
REMOVED_DIR="$(workspaces.input.path)/dir-to-be-cleaned-up"
KEPT_DIR="$(workspaces.input.path)/dir-to-be-kept"
```

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+              REMOVED_DIR="$(workspaces.input.path)/dir-to-be-cleaned-up"
+              # Check if dir-to-be-cleaned-up deleted, it prints non-targeted file contents
+              [ -d "$REMOVED_DIR" ] && echo "Something wrong directory $REMOVED_DIR found." ||
```
Comment (mmalina) [inline catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml:61]: What is the `||` at the end for? Also, the `echo` should clearly say what the dir is expected to be deleted, but was still there. Also, I would suggest to use a standard `if` here - it's more readable.

And the check of the kept dir/file should be also more explicit - right now if it was not there, it would just fail on a file not found error. It would be nicer if it prints some message.

So I'd suggest something like this:
```sh
if [ -d "$REMOVED_DIR" ]
then
    echo Dir $REMOVED_DIR is expected to be deleted, but still exists
    exit 1
fi

if [ ! -d "$KEPT_DIR" ] || [ ! -f "${KEPT_DIR}/cleanup_check.txt" ]
then
    echo File ${KEPT_DIR}/cleanup_check.txt is expected to exist, but it does not
    exit 1
fi
```

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml
Comment (mmalina) [inline catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml:1]: You could also add another test for the empty `subdirectory` parameter scenario and/or for when the parameter points to a non-existing directory. In both cases, the task should succeed.

**Comment by johnbieren** [comment]
> The commit title is still wrong and it needs rebase

I think the proper prefix for this would be `tests(HACBS-2231): ...`

**Comment by mmalina** [comment]
> I think the proper prefix for this would be `tests(HACBS-2231): ...`

Looking at `git log`, it seems we never had `tests(` or `test(` in the commit titles in this repo (probably because we never really had any tests here). But in the release-service repo, I found multiple occurrences of `test(`. And while [the specification](https://www.conventionalcommits.org/en/v1.0.0/#specification) says you can use anything you want, it does give `test` as an example. So it would probably make sense to use `test(HACBS-2231)` here?

**Comment by johnbieren** [comment]
> > I think the proper prefix for this would be `tests(HACBS-2231): ...`
> 
> Looking at `git log`, it seems we never had `tests(` or `test(` in the commit titles in this repo (probably because we never really had any tests here). But in the release-service repo, I found multiple occurrences of `test(`. And while [the specification](https://www.conventionalcommits.org/en/v1.0.0/#specification) says you can use anything you want, it does give `test` as an example. So it would probably make sense to use `test(HACBS-2231)` here?

makes sense to me

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml
Comment (happybhati) [reply catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml:1]: I have added the test as you requested, but I need your feedback on this.
`test-cleanup-workspace-no-subdirectory.yaml`

**Comment by johnbieren** [comment]
Can you make the `Add` in your commit title `add` https://www.conventionalcommits.org/en/v1.0.0/

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace-no-subdirectory.yaml
Comment (johnbieren) [inline catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace-no-subdirectory.yaml:1]: I think it might be nice to separate these tests out into their own files

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      taskSpec:
+        workspaces:
+          - name: input
+        steps:
+          - name: put-content-in-workspace
+            image: quay.io/hacbs-release/release-utils:5b1a1cd9fd68625cab5573ce62e0d87e6f93f341
+            script: |
+              #!/usr/bin/env sh
+              
+              INPUT_PATH="$(workspaces.input.path)"
```
Comment (johnbieren) [inline catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml:26]: I think Martin was looking for this variable to be removed and instead use 
```
REMOVED_DIR="$(workspaces.input.path)/dir-to-be-cleaned-up"
KEPT_DIR="$(workspaces.input.path)/dir-to-be-kept"
```

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+              mkdir -p $INPUT_PATH/dir-to-be-cleaned-up/
+              echo "This file and dir will be deleted" >> $INPUT_PATH/dir-to-be-cleaned-up/cleanup_test.txt
+              mkdir -p $INPUT_PATH/dir-to-be-kept/
+              echo "Test successful" >> $INPUT_PATH/dir-to-be-kept/cleanup_check.txt
+              ls -R $INPUT_PATH/
```
Comment (johnbieren) [inline catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml:31]: this is probably nice as you write the task, but I don't think it needs to be in the final version. We should have some faith that `echo foo >> file` works

File: catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml (hunk 4 above)
Comment (johnbieren) [reply catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml:26]: I see you did in the check task, might as well do it here too
